### PR TITLE
Fix EMCC_FORCE_STDLIBS and re-enable tests for it

### PR DESCRIPTION
**Pull request by sbc100**
Refactored the existing test and re-enabled it.  This had
bit rotted in the mean time which meant that EMCC_FORCE_STDLIBS=1
was getting duplicate symbols.

Fixes: #10571
Fixes: #11703

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -823,7 +822,7 @@ def get_files(self):
         filenames=['vfprintf.c'])
 
   def can_build(self):
-    return super(libprintf_long_double, self).can_build() and shared.Settings.WASM_BACKEND
+    return super(libprintf_long_double, self).can_build() and shared.Settings.WASM_BACKEND and shared.Settings.PRINTF_LONG_DOUBLE
```
Comment (kripken) [inline tools/system_libs.py:825]: this doesn't match my understanding of `can_build` (i think this should go under `can_use` instead). but maybe I have it wrong? why was this change needed?

File: tests/test_core.py [hunk 2]
```diff
@@ -4391,44 +4391,45 @@ def test_dylink_postsets_chunking(self):
...
-        print('(seeing expected fail)')
-        passed = False
-        assertion = 'build the MAIN_MODULE with EMCC_FORCE_STDLIBS=1 in the environment'
-        if self.get_setting('ASSERTIONS'):
-          self.assertContained(assertion, str(e))
-        else:
-          self.assertNotContained(assertion, str(e))
-      assert passed == expect_pass, ['saw', passed, 'but expected', expect_pass]
+    if assertions is not None:
+      if assertions:
```
Comment (kripken) [inline tests/test_core.py:4404]: `set_setting('ASSERTIONS', int(assertions))` on one line?

**Comment by sbc100** [comment]
Reviving this old PR.. PTAL

File: tools/system_libs.py [hunk 3]
```diff
@@ -894,6 +896,9 @@ def get_files(self):
     network_dir = shared.path_from_root('system', 'lib', 'libc', 'musl', 'src', 'network')
     return [os.path.join(network_dir, x) for x in LIBC_SOCKETS]
 
+  def can_use(self):
+    return super(libsockets, self).can_use() and not settings.PROXY_POSIX_SOCKETS
```
Comment (kripken) [inline tools/system_libs.py:900]: is this right? I think the one after it is, but am not sure here.

**Review by kripken — COMMENTED** [review]
lgtm % comment

File: tools/system_libs.py (hunk 3 above)
Comment (sbc100) [reply tools/system_libs.py:900]: Yes, I believe the `libsockets` and `libsockets_proxy` provide the same API so they cannot be included at the same time without generating duplicate symbol errors.  

You can see here are they are used (under normal circumstances) in a mutually exclusive manor:
https://github.com/emscripten-core/emscripten/blob/d35de0bf3572dfe9759ac3c7e56b9ce5983f68ac/tools/system_libs.py#L1621-L1624

Comment (kripken) [reply tools/system_libs.py:900]: Thanks, now I see. lgtm!

**Review by kripken — APPROVED** [review]
